packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -434,18 +434,20 @@ export function HierarchicalTermSelector( { slug } ) {
 				) }
 			</div>
 			{ ! loading && hasCreateAction && (
-				<Button
-					onClick={ onToggleForm }
-					className="editor-post-taxonomies__hierarchical-terms-add"
-					aria-expanded={ showForm }
-					variant="link"
-				>
-					{ newTermButtonLabel }
-				</Button>
+				<FlexItem>
+					<Button
+						onClick={ onToggleForm }
+						className="editor-post-taxonomies__hierarchical-terms-add"
+						aria-expanded={ showForm }
+						variant="link"
+					>
+						{ newTermButtonLabel }
+					</Button>
+				</FlexItem>
 			) }
 			{ showForm && (
 				<form onSubmit={ onAddTerm }>
-					<VStack>
+					<Flex direction="column" gap="4">
```

Comment:
Should we prefer a `<div />` (or any other HTML element) with custom CSS flex styles here, as mentioned in https://github.com/WordPress/gutenberg/issues/43085 ? Or should we continue using `Flex` for now, and look into migrating away from it at a later point? (cc @mirka )

Reply:
Seeing that we may use `FlexItem` as suggested in https://github.com/WordPress/gutenberg/pull/47515#discussion_r1093269132, we can probably also keep using `Flex` for the time being.

We can have a look at "dismissing" `Flex` and its subcomponents at a later point

Reply:
I guess the pro/cons to using `Flex` are similar to Tailwind, where you get dramatically less CSS data in the final output in return for having to learn a new set of styling conventions. I don't have too strong of an opinion one way or the other, but I slightly lean towards using `Flex` for now because it will probably be easier to refactor once we are ready to move to a better spacing component (#43423).

Reply:
Sounds good!